### PR TITLE
Add backend APIs for issuing U2F requests and processing U2F responses.

### PR DESCRIPTION
**Pull request by xiaoliu1988**
Hi @leshi, please review this change!

Endpoint APIs include:
1. return all registered security keys;
2. issue a registration request;
3. process a registration reponse;
4. issue a signature request;
5. process a signature response.

### CODE REVIEW

**Comment by xiaoliu1988** [comment]
I signed it!

File: u2f-gae-demo/src/com/google/u2f/gaedemo/endpoints/U2FRequestHandler.java [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+        dataStore.getSignSessionData(responseDataJson.get("sessionId").toString()).getAccountName();
+
+    if (!currentUser.equals(expectedUser)) {
+      throw new ServletException("Cross-site request prohibited");
+    }
+
+    SignResponse signResponse =
+        new SignResponse(
+            responseDataJson.get("keyHandle").toString(),
+            responseDataJson.get("signatureData").toString(),
```
Comment (leshi) [inline u2f-gae-demo/src/com/google/u2f/gaedemo/endpoints/U2FRequestHandler.java]: Making a constant for these strings would be nice.

File: u2f-gae-demo/src/com/google/u2f/gaedemo/endpoints/U2FRequestHandler.java [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+    if (user == null) {
+      throw new OAuthRequestException("user is not authenticated");
+    }
+
+    if (u2fServer == null) {
+      initU2fServer();
+    }
+
+    u2fServer.removeSecurityKey(user.getEmail(), Hex.decodeHex(publicKey.toCharArray()));
+    return new String[] {"OK"};
```
Comment (leshi) [inline u2f-gae-demo/src/com/google/u2f/gaedemo/endpoints/U2FRequestHandler.java]: where does this go?

File: u2f-gae-demo/src/com/google/u2f/gaedemo/endpoints/U2FRequestHandler.java [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+      throw new OAuthRequestException("user is not authenticated");
+    }
+
+    JsonObject responseDataJson = (JsonObject) new JsonParser().parse(responseData);
+    String currentUser = user.getEmail();
+    String expectedUser =
+        dataStore.getSignSessionData(responseDataJson.get("sessionId").toString()).getAccountName();
+
+    if (!currentUser.equals(expectedUser)) {
+      throw new ServletException("Cross-site request prohibited");
```
Comment (xiaoliu1988) [inline u2f-gae-demo/src/com/google/u2f/gaedemo/endpoints/U2FRequestHandler.java]: Done.

File: u2f-gae-demo/src/com/google/u2f/gaedemo/endpoints/U2FRequestHandler.java (hunk 1 above)
Comment (xiaoliu1988) [inline u2f-gae-demo/src/com/google/u2f/gaedemo/endpoints/U2FRequestHandler.java]: done

File: u2f-gae-demo/src/com/google/u2f/gaedemo/endpoints/U2FRequestHandler.java (hunk 2 above)
Comment (xiaoliu1988) [inline u2f-gae-demo/src/com/google/u2f/gaedemo/endpoints/U2FRequestHandler.java]: Returning a simple type such as String or int is not supported. The return value needs to be a JavaBean (non-parametrized), an array or a Collection. 
https://cloud.google.com/appengine/docs/standard/java/endpoints/parameter-and-return-types

I chose to use simple array for all method return types; and for this method specifically, I thought using "OK" to represent the successful completion of that operation, the caller can choose to examine this status.

**Comment by xiaoliu1988** [comment]
This one will be closed once #152 is reviewed.

**Pull request closed without merging**